.gitlab-ci.yml

```diff
@@ -0,0 +1,139 @@
+default:
+  tags:
+    - macos-arm64
+
+include:
+  - project: hpc/gitlab-upload-logs
+    file: enable-upload.yml
+
+variables:
+  GIT_STRATEGY: none
+  DESIRED_CMAKE_VERSION: 3.25.0-rc2
+
+stages:
+  - clean workspace
+  - prerequisites
+  - build
+  - test
+
+# see https://gitlab.com/gitlab-org/gitlab/-/issues/263401 for why we specify the flags like this now
+# 130 characters seems to be the point at which jobs refuse to run
+.matrix:
+  parallel:
+    matrix:
+      - build_mode: cmake
+        cmake_coreneuron: "ON"
+        cmake_interviews: "OFF"
+        cmake_rx3d: "OFF"
+        sanitizer: address
+
+clean_workspace:
+  stage: clean workspace
+  variables:
+    GIT_STRATEGY: clone
+  script:
+    - echo "Workspace clean"
+
+set_homebrew_variables:
+  stage: prerequisites
+  extends: .matrix
+  script:
+    - echo CMAKE_BUILD_PARALLEL_LEVEL=3 >> build.env
+    - echo CTEST_PARALLEL_LEVEL=3 >> build.env
+    - echo CI_OS_NAME=osx >> build.env
```

Comment:
I think this is only needed for the `test_wheels` script, which we're not running here